concordium-node/src/connection/mod.rs

```diff
@@ -639,28 +639,19 @@ impl Connection {
         let peer_list_resp = match self.handler.peer_type() {
             PeerType::Bootstrapper => {
                 // select random nodes that are post-handshake
-                let get_random_nodes = |partition: bool| -> Fallible<Vec<P2PPeer>> {
+                let get_random_nodes = || -> Fallible<Vec<P2PPeer>> {
                     Ok(read_or_die!(self.handler.buckets())
                         .get_random_nodes(
                             requestor,
                             self.handler.config.bootstrapper_peer_list_size,
                             &nets,
-                            partition,
                         )
                         .iter()
                         .filter_map(RemotePeer::peer)
                         .collect())
                 };
 
-                #[cfg(not(feature = "malicious_testing"))]
-                let random_nodes = get_random_nodes(false)?;
-                #[cfg(feature = "malicious_testing")]
-                let random_nodes = match self.handler.config.partition_network_for_time {
-                    Some(time) if (self.handler.get_uptime() as usize) < time => {
-                        get_random_nodes(true)?
-                    }
-                    _ => get_random_nodes(false)?,
-                };
+                let random_nodes = get_random_nodes()?;
```

Comment:
There is no point declaring the `get_random_nodes` closure anymore. Just
```rust
let random_nodes = read_or_die!(self.handler.buckets())
                        .get_random_nodes(
                            requestor,
                            self.handler.config.bootstrapper_peer_list_size,
                            &nets,
                            partition,
                        )
                        .iter()
                        .filter_map(RemotePeer::peer)
                        .collect();
```
should be fine.

Reply:
True. I have simplified the code.